base_jsonify/models/models.py

```diff
@@ -54,31 +58,52 @@ def jsonify(self, parser):
             res = {}
             for field in parser:
                 field_name, json_key, subparser = self.__parse_field(field)
-                field_type = rec._fields[field_name].type
                 if subparser:
-                    if field_type in ("one2many", "many2many"):
-                        res[json_key] = rec[field_name].jsonify(subparser)
-                    elif field_type in ("many2one", "reference"):
-                        if rec[field_name]:
-                            res[json_key] = rec[field_name].jsonify(subparser)[0]
-                        else:
-                            res[json_key] = None
-                    else:
-                        raise UserError(_("Wrong parser configuration"))
+                    res[json_key] = rec._jsonify_value_subparser(field_name, subparser)
                 else:
-                    value = rec[field_name]
-                    if value is False and field_type != "boolean":
-                        value = None
-                    elif field_type == "date":
-                        value = fields.Date.to_date(value).isoformat()
-                    elif field_type == "datetime":
-                        # Ensures value is a datetime
-                        value = fields.Datetime.to_datetime(value)
-                        # Get the timestamp converted to the client's timezone.
-                        # This call also add the tzinfo into the datetime
-                        # object
-                        value = fields.Datetime.context_timestamp(rec, value)
-                        value = value.isoformat()
-                    res[json_key] = value
+                    res[json_key] = rec._jsonify_value(field_name)
             result.append(res)
         return result
+
+    def _jsonify_value(self, field_name):
+        field_type = self._fields[field_name].type
+        value = self[field_name]
+        # TODO: we should get default by field (eg: char field -> "")
+        if value is False and field_type != "boolean":
+            value = None
+        elif field_type == "date":
+            value = fields.Date.to_date(value).isoformat()
+        elif field_type == "datetime":
+            # Ensures value is a datetime
+            value = fields.Datetime.to_datetime(value)
+            # Get the timestamp converted to the client's timezone.
+            # This call also add the tzinfo into the datetime
+            # object
+            value = fields.Datetime.context_timestamp(self, value)
+            value = value.isoformat()
+        return value
+
+    def _jsonify_value_subparser(self, field_name, subparser):
+        value = None
+        rec_value = self[field_name]
```

Comment:
The first two cases don't use this value, so I think it's better to move it into the `else:` block below. That way you can also make "fake" computed fields that don't correspond to any field on the model. If you try that now it crashes at this line.